src/core/sync/mutex.d

```diff
@@ -131,6 +131,11 @@ nothrow:
      */
     @trusted void lock()
     {
+        lock_final();
+    }
+
+    final @trusted void lock_final()
```

Comment:
That's fairly ugly to change the API, and you can't use package protection here.
Maybe we just copy over the implementation to the GC?